lib/repository.py

```diff
@@ -56,6 +56,19 @@ def get_git_repository(remote_repo_url, parent_dir_path):
                                         proxy=CONF.get('http_proxy'))
 
 
+def get_svn_repository(remote_repo_url, repo_path):
+    """
+    Get a local subversion repository located in a directory.
+    If it does not exist, check it out from the specified URL.
+    """
+    if os.path.exists(repo_path):
+        return SvnRepository(remote_repo_url, repo_path)
+    else:
+        # TODO: setup HTTP proxy by editing ~/.subversion/servers
```

Comment:
As SVN has a global config for this, IMHO this should be done before building anything instead of doing it specifically for the gcc build. We may (1) automate it and do it before building anything in host_os.py or (2) require user to do it manually and document it. 1 is better, obviously, but if the kernel team already uses our build scripts to build gcc, they already have setup this manually, so I do not see this as a blocker.